ui/src/main/java/org/pentaho/di/ui/i18n/MessagesSourceCrawler.java

```diff
@@ -88,115 +149,77 @@
    */
   private List<SourceCrawlerXMLFolder> xmlFolders;
 
-  private Pattern packagePattern;
-  private Pattern importPattern;
-  private Pattern importMessagesPattern;
-  private Pattern stringPkgPattern;
-  private Pattern classPkgPattern;
+  private Pattern packagePattern = Pattern.compile( PACKAGE_PATTERN );
+  private Pattern importPattern = Pattern.compile( IMPORT_PATTERN );
+  private Pattern importMessagesPattern = Pattern.compile( IMPORT_MESSAGES_PATTERN );
+  private Pattern stringPkgPattern = Pattern.compile( STRING_PKG_PATTERN );
+  private Pattern classPkgPattern = Pattern.compile( CLASS_PKG_PATTERN );
 
   private LogChannelInterface log;
 
   /**
-   * @param sourceDirectories
-   *          The source directories to crawl through
-   * @param singleMessagesFile
-   *          the messages file if there is only one, otherwise: null
+   * @param sourceDirectories  The source directories to crawl through
+   * @param singleMessagesFile the messages file if there is only one, otherwise: null
    */
   public MessagesSourceCrawler( LogChannelInterface log, List<String> sourceDirectories,
-    String singleMessagesFile, List<SourceCrawlerXMLFolder> xmlFolders ) {
+                                String singleMessagesFile, List<SourceCrawlerXMLFolder> xmlFolders ) {
     super();
     this.log = log;
-    this.sourceDirectories = sourceDirectories;
+    this.sourceDirectories = ( null != sourceDirectories ) ? sourceDirectories : new ArrayList<>();
     this.singleMessagesFile = singleMessagesFile;
-    this.filesToAvoid = new ArrayList<String>();
-    this.xmlFolders = xmlFolders;
-
-    this.sourcePackageOccurrences = new HashMap<String, Map<String, List<KeyOccurrence>>>();
-
-    packagePattern = Pattern.compile( "^\\s*package .*;[ \t]*$" );
-    importPattern = Pattern.compile( "^\\s*import [a-z\\._0-9]*\\.[A-Z].*;[ \t]*$" );
-    importMessagesPattern = Pattern.compile( "^\\s*import [a-z\\._0-9]*\\.Messages;[ \t]*$" );
-    stringPkgPattern = Pattern.compile( "^.*private static String PKG.*=.*$" );
-    classPkgPattern = Pattern.compile( "^.*private static Class.*\\sPKG\\s*=.*$" );
-  }
-
-  /**
-   * @return The source directories to crawl through
-   */
-  public List<String> getSourceDirectories() {
```

Comment:
Don't remove public methods from a public class that someone may have overridden. Remember, this is open source code. If these are not necessary anymore please deprecate them.

Reply:
It was only moved... it's now near line 740 :-)